build-tests/sparo-real-repo-test/src/start-test.ts

```diff
@@ -0,0 +1,113 @@
+import * as path from 'path';
+import { FileSystem } from '@rushstack/node-core-library';
+import type { IRunScriptOptions } from '@rushstack/heft';
+import {
+  ICommandDefinition,
+  executeCommandsAndCollectOutputs,
+  updateOrCompareOutputs
+} from 'build-test-utilities';
+
+/**
+ * This build test is highly inspired by the build test for api-extractor in rushstack.
+ */
+export async function runAsync(runScriptOptions: IRunScriptOptions): Promise<void> {
+  const {
+    heftTaskSession: {
+      logger,
+      parameters: { production }
+    },
+    heftConfiguration: { buildFolderPath }
+  } = runScriptOptions;
+
+  const temporaryDirectory: string = path.resolve(buildFolderPath, 'temp');
+  const testRepoURL: string = 'https://github.com/tiktok/sparo.git';
```

Comment:
In order to ensure deterministic behavior, the test should run against a static input.

Suggest to create a GitHub branch like `/test-artifacts/sparo-real-repo-test` and have the test clone that specific branch.

Reply:
test-artifacts/sparo-real-repo-test created.